app/packages/core/src/components/ColorModal/JSONViewer.tsx

```diff
@@ -57,14 +71,27 @@ const JSONViewer: React.FC = () => {
     });
   };
 
-  useEffect(() => {
+  useLayoutEffect(() => {
     setData(setting);
+    console.info("setData");
+    if (ref?.current) {
+      console.log(ref?.current);
```

Comment:
I am trying to dispatch an event whenever the json viewer has updated the data (there is a loading status). Tested in the browser, ref.current is always none. @sashankaryal suggestion on other ref that I can dispatch the event?

Reply:
Add `ref` to the effect dependency list and it should work

Reply:
it works now. Thanks, @sashankaryal !